### PR TITLE
Textlabel fixes

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: src/widgets/textlabel.h [hunk 1]
```diff
@@ -2,7 +2,6 @@
 #define TEXTLABEL_H
 
 #include <QLabel>
-#include <QMouseEvent>
```
Comment (shinnok) [inline src/widgets/textlabel.h]: QMouseEvent is used in the header for the mouseReleaseEvent override, on certain toolchains this is a build error.

File: src/widgets/archivelistwidget.cpp [hunk 2]
```diff
@@ -4,6 +4,7 @@
 #include "ui_archiveitemwidget.h"
 
 #include <QMessageBox>
+#include <QKeyEvent>
```
Comment (shinnok) [inline src/widgets/archivelistwidget.cpp]: The Title of the PR says Textlabel fixes. Can you break unrelated changes from now on into separate PRs?

Comment (gperciva) [inline src/widgets/archivelistwidget.cpp]: This was related to the `QMouseEvent` removal from `textlabel.h`.  If I didn't remove that, the other files compiled fine (presumably because they implicitly included `QMouseEvent`, although I'm not certain how that would supply `QKeyEvent` since according to the Qt docs it's not a direct inherited class).  I therefore thought it was related to this PR, since a PR should never leave the tree in an un-compilable state.

File: src/widgets/textlabel.h (hunk 1 above)
Comment (gperciva) [inline src/widgets/textlabel.h]: which toolchains?  That sounds really weird, since `qwidget.h` gives a forward declaration of `class QMouseEvent`.  Which toolchain can't deal with a pointer to a forward declared class?

File: src/widgets/archivelistwidget.cpp (hunk 2 above)
Comment (gperciva) [inline src/widgets/archivelistwidget.cpp]: That said, there's certainly no problem with adding `QKeyEvent` first, independent of whether `QMouseEvent` is removed from `textlabel.h`.  So I'll make a separate PR for that now.  :)

Comment (gperciva) [inline src/widgets/archivelistwidget.cpp]: submitted as #74

File: src/widgets/textlabel.h (hunk 1 above)
Comment (shinnok) [inline src/widgets/textlabel.h]: Can't remember off the top of my head which version of Qt and on which platform. Maybe it will come up again or not, will see when I get to the next round of platform testing. Merged.
